.github/workflows/release.yml

```diff
@@ -0,0 +1,49 @@
+name: Publish release
```

Comment:
This replaces the `publish_code.yml` workflow.
It also replaces the legacy upload using PyPI tokens to use the new trusted publishers mechanism.
This still has to be configured on the PyPI side after this PR is merged.